library/std/src/sys/wasix/process/process_wasix.rs

```diff
@@ -2,15 +2,33 @@
 use crate::fmt;
 use crate::io;
 use crate::num::NonZeroI32;
+use crate::os::fd::FromRawFd;
+use crate::sys::pipe::AnonPipe;
 use crate::sys::process::process_common::*;
 use crate::sys::{unsupported, unsupported_err};
-use core::ffi::NonZero_c_int;
+use core::ffi::{CStr, NonZero_c_int};
+use wasi::{JoinStatus, JoinStatusType};
 
 use crate::io::ErrorKind;
 
 use libc::{c_int, pid_t};
 
-pub use crate::sys::{cvt, cvt_r, cvt_nz};
+pub use crate::sys::{cvt, cvt_nz, cvt_r};
+
+fn to_anon_pipe(option_fd: wasi::OptionFd) -> Option<AnonPipe> {
+    match wasi::Option::from(option_fd.tag) {
+        wasi::OPTION_SOME => Some(unsafe { AnonPipe::from_raw_fd(option_fd.u.some as i32) }),
+        _ => None,
+    }
+}
+
+fn to_wasi_mode(input: &Stdio) -> wasi::StdioMode {
+    match input {
+        Stdio::Inherit => wasi::STDIO_MODE_INHERIT,
+        Stdio::MakePipe => wasi::STDIO_MODE_PIPED,
+        _ => wasi::STDIO_MODE_NULL,
```

Comment:
Should we support other modes here?

Reply:
These are actually all the available modes at the moment, so nothing to do on our side.